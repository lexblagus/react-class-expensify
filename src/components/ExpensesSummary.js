import React from 'react';
import selectExpenses from '../selectors/expenses';
import expensesTotal from '../selectors/expenses-total';
import { connect } from 'react-redux';

export const ExpensesSummary = ({qtt,tot}) => (
	<div>
		<h3 className="title is-3">
			List
			<span className="title is-4">
				{' '}
				with
				{' '}
				{qtt}
				{' '}
				item{qtt>1?'s':''}
				{' '}
				$ {tot}
			</span>
		</h3>
	</div>
);

const mapStateToProps = (state) => ({
		'tot': expensesTotal(selectExpenses(state.expenses,state.filters)),
		'qtt': selectExpenses(state.expenses,state.filters).length
});


export default connect(mapStateToProps)(ExpensesSummary);

//export default ExpensesSummary;