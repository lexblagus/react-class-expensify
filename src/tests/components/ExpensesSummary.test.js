import React from 'react';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
import ExpensesSummary from '../../components/ExpensesSummary.js';
import dummyStore02 from '../fixtures/dummyStore02';
import { filters_0, filters_1 } from '../fixtures/dummyFilters';

test('should render ExpensesSummary with 1 item',()=>{
	const wrapper = shallow(
		<ExpensesSummary
			qtt={1}
			tot={1}
		/>
	);
	expect( wrapper ).toMatchSnapshot();
});


test('should render ExpensesSummary with plural items',()=>{
	const wrapper = shallow(
		<ExpensesSummary
			qtt={999}
			tot={666}
		/>
	);
	expect( wrapper ).toMatchSnapshot();
});
