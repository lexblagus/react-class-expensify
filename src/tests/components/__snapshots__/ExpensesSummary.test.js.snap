// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with 1 item 1`] = `
<div>
  <h3
    className="title is-3"
  >
    List
    <span
      className="title is-4"
    >
       
      with
       
      1
       
      item
       
      $ 
      1
    </span>
  </h3>
</div>
`;

exports[`should render ExpensesSummary with plural items 1`] = `
<div>
  <h3
    className="title is-3"
  >
    List
    <span
      className="title is-4"
    >
       
      with
       
      999
       
      item
      s
       
      $ 
      666
    </span>
  </h3>
</div>
`;
